ocs_ci/ocs/cluster.py

```diff
@@ -3188,3 +3190,81 @@ def get_mon_quorum_ranks():
     for rank in list(out["quorum"]):
         mon_quorum_ranks[list(out["quorum_names"])[rank]] = rank
     return mon_quorum_ranks
+
+
+def client_cluster_health_check():
+    """
+    Check the client cluster health.
+
+    The function will check the following:
+    1. Wait for the cluster connectivity
+    2. Wait for the nodes to be in a Ready state
+    3. Checking that there are no extra Ceph pods on the cluster
+    4. Wait for the pods to be running in the cluster namespace
+    5. Checking that the storageclient is connected
+
+    Raises:
+        ResourceWrongStatusException: In case not all the nodes are ready, not all the pods are running, or
+            the storageclient is not connected
+        CephHealthException: In case there are extra Ceph pods on the cluster
+
+    """
+    wait_for_cluster_connectivity(tries=120, delay=5)
+    logger.info("Checking the cluster health")
+    wait_for_nodes_status(timeout=300, sleep=10)
+
+    logger.info("Checking that there are no extra Ceph pods on the cluster")
+    mon_pods = pod.get_mon_pods()
+    if mon_pods:
+        raise exceptions.CephHealthException(
+            "The client Cluster shouldn't have any mon pods!"
+        )
+    osd_pods = pod.get_osd_pods()
+    if osd_pods:
+        raise exceptions.CephHealthException(
+            "The client Cluster shouldn't have any osd pods!"
+        )
+    mds_pods = pod.get_mds_pods()
+    if mds_pods:
+        raise exceptions.CephHealthException(
+            "The client Cluster shouldn't have any mds pods!"
+        )
+
+    logger.info("Wait for the pods to be running")
+    res = wait_for_pods_to_be_running(timeout=300, sleep=20)
+    if not res:
+        raise ResourceWrongStatusException("Not all the pods in running state")
+
+    logger.info("Checking that the storageclient is connected")
+    sc_obj = OCP(
+        kind=constants.STORAGECLIENT, namespace=config.ENV_DATA["cluster_namespace"]
+    )
+    sc_obj.wait_for_resource(
+        resource_name=constants.DEFAULT_STORAGE_CLIENT,
```

Comment:
This would be better to take from config. There can be storage_client_name in ENV_DATA in https://github.com/red-hat-storage/ocs-ci/blob/master/ocs_ci/framework/conf/default_config.yaml#L146 so that it doesn't have to be specified in platform configs. It can be obtained for correct context with `config.cluster_ctx.ENV_DATA.get("storage_client_name")`.
In future we could want to test other storageclient names and it can be allowed with this change.

Reply:
Okay, I will add it to the config.